streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -645,7 +650,12 @@ private int restoreChangelog(final Task task, final ChangelogMetadata changelogM
 
         if (numRecords != 0) {
             final List<ConsumerRecord<byte[], byte[]>> records = changelogMetadata.bufferedRecords.subList(0, numRecords);
-            stateManager.restore(storeMetadata, records);
+            final OptionalLong optionalLag = restoreConsumer.currentLag(partition);
+            Long currentLag = null;
```

Comment:
just a nit, feel free to ignore, but we can simplify this a bit by just passing the `OptionalLong` directly into `#restore`. I know you're not "supposed" to use Optional as an argument but in this case it lets us do all the special-case handling in one place, and makes it clear that the currentLag may be empty/uninitialized when we go to use it in `#restore`